### PR TITLE
ref(hc): Use new access service

### PR DESCRIPTION
**Pull request by corps**
First step for more performant and resilient auth is getting AuthProviderReplica and AuthIdentityReplica used in the access layer.  Extracting logic here so that the service layer is split between silos using the correct model.

Next steps are to further remove cross silo RPCs with more replicated models.  First step of many.

### CODE REVIEW

File: src/sentry/migrations/0542_add_last_verified_auth_ident_replica.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
```
Comment (GabeVillalobos) [inline src/sentry/migrations/0542_add_last_verified_auth_ident_replica.py:19]: Is it possible to decouple this migration from the access service logic?

**Comment by GabeVillalobos** [comment]
Changes look good to me! Just unsure if we want to bundle a migration with the access logic changes if possible.

File: src/sentry/migrations/0542_add_last_verified_auth_ident_replica.py (hunk 1 above)
Comment (corps) [reply src/sentry/migrations/0542_add_last_verified_auth_ident_replica.py:19]: Sure, I'll extract

Comment (corps) [reply src/sentry/migrations/0542_add_last_verified_auth_ident_replica.py:19]: https://github.com/getsentry/sentry/pull/55783

**Review by GabeVillalobos — APPROVED** [review]

File: src/sentry/auth/access.py [hunk 2]
```diff
@@ -1130,7 +1119,7 @@ def from_user(
         return organizationless_access(user, is_superuser)
 
     try:
-        om = get_cached_organization_member(user.id, organization.id)
+        om = OrganizationMember.objects.get(user_id=user.id, organization_id=organization.id)
```
Comment (markstory) [inline src/sentry/auth/access.py:1122]: Do we not get access `from_user` in control silo?

File: src/sentry/models/authidentity.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 class AuthIdentity(ReplicatedControlModel):
     __relocation_scope__ = RelocationScope.Global
     category = OutboxCategory.AUTH_IDENTITY_UPDATE
+    replication_version = 2
```
Comment (markstory) [inline src/sentry/models/authidentity.py:20]: Should we deploy workers for the backlog this is going to generate?

File: src/sentry/services/hybrid_cloud/access/impl.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        self, auth_provider: RpcAuthProvider, member: RpcOrganizationMemberSummary
+    ) -> bool:
+        """If an owner is trying to gain access, allow bypassing SSO if there are no
+        other owners with SSO enabled.
+        """
+
+        # Get more org role related data into control to reduce this inter silo rpc stuff....
+        org_roles = organization_service.get_all_org_roles(
+            member_id=member.id, organization_id=member.organization_id
+        )
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/access/impl.py:64]: Is related to team roles?

File: src/sentry/services/hybrid_cloud/access/service.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+def impl_by_control_resources() -> AccessService:
+    from sentry.services.hybrid_cloud.access.impl import ControlAccessService
+
+    return ControlAccessService()
+
+
+access_service: AccessService = silo_mode_delegation(
+    {
+        SiloMode.REGION: impl_by_region_resources,
+        SiloMode.CONTROL: impl_by_control_resources,
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/access/service.py:135]: Do we have any way to validate that the two implementations behave the same given a consistent starting point with replicated records?

File: src/sentry/services/hybrid_cloud/access/impl.py (hunk 4 above)
Comment (corps) [reply src/sentry/services/hybrid_cloud/access/impl.py:64]: Yup, the intention will be to replicate some aspect of team roles into control.

File: src/sentry/services/hybrid_cloud/access/service.py (hunk 5 above)
Comment (corps) [reply src/sentry/services/hybrid_cloud/access/service.py:135]: Yes, in the sense that test_access already does this by using the same suite for all modes.

**Pull request merged**